backend/controller/controller.go

```diff
@@ -98,10 +98,14 @@ type Service struct {
 	deploymentReservationTimeout time.Duration
 	artefactChunkSize            int
 	key                          model.ControllerKey
+	ctx                          context.Context
 
 	clientsMu sync.Mutex
 	// Map from endpoint to client.
 	clients map[string]clients
+	logger  *log.Logger
+	// Map from deployment key to logger.
+	deploymentLoggers map[model.DeploymentKey]*log.Logger
```

Comment:
Is this ok? Since we can have multiple deployments it seems like we need multiple loggers for them. This part felt a bit odd to me.